### ISSUE TITLE
RStudio crashes due to missing glibc if `r_pkgs` is `NULL`

### ISSUE THREAD

**Issue by eliocamp** [b-rodrigues/rix#214]
This nix config installs everything fine, but when I open RStudio, it crashes with ```/usr/lib/rstudio/resources/app/bin/rsession: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /nix/store/9lzp5kbn51k4v1kdyh8irwmin4kri72k-R-4.3.2/lib/R/lib/libR.so)```

```r
library(rix)
rix(r_ver = "4.3.2",
    r_pkgs = NULL,
    system_pkgs = NULL,
    git_pkgs = NULL,
    ide = "rstudio",
    project_path = ".",
    overwrite = TRUE,
    print = TRUE)

```

![image](https://github.com/b-rodrigues/rix/assets/8617595/fba72962-dc7d-40d4-97f9-e51af68c6f41)

However this

```r
rix(r_ver = "4.3.2",
    r_pkgs = c("dplyr", "ggplot2"),
    system_pkgs = NULL,
    git_pkgs = NULL,
    ide = "rstudio",
    project_path = ".",
    overwrite = TRUE,
    print = TRUE)
```

runs fine.

**Comment by b-rodrigues** [comment]
probably it tries to load your local packages. Try rix::rix_init() and my guess is that it should work because it won't be looking for packages in your system library

**Comment by eliocamp** [comment]
I tried after `rix_init()` and the same thing happened.

**Comment by philipp-baumann** [comment]
@eliocamp @b-rodrigues i think i had something similar on my rocky linux miniserver; looks almost like glibc from the system is used. 

guess its the same when you start rstudio via `nix-shell --pure`?

**Comment by philipp-baumann** [comment]
Thanks a lot for reporting the bug @eliocamp

The problem is that currently no `r_pkgs` are supplied, resulting nix expression does not include any rstudioWrapper, which it should. 

```r
library(rix)
rix(r_ver = "4.3.2",
    r_pkgs = NULL,
    system_pkgs = NULL,
    git_pkgs = NULL,
    ide = "rstudio",
    project_path = ".",
    overwrite = TRUE,
    print = TRUE)
```

does not build rstudio via nix and nix shell does not make it available (so starting it with `nix-shell --pure` will not find rstudio, otherwise you will open the system one like in the screenshot.

```sh
# This file was generated by the {rix} R package v0.7.1 on 2024-06-24
# with following call:
# >rix(r_ver = "219f896bdf192721446db4fedc338eebf732057d",
#  > r_pkgs = NULL,
#  > system_pkgs = NULL,
#  > git_pkgs = NULL,
#  > ide = "rstudio",
#  > project_path = "_rstudio",
#  > overwrite = TRUE,
#  > print = TRUE)
# It uses nixpkgs' revision 219f896bdf192721446db4fedc338eebf732057d for reproducibility purposes
# which will install R version 4.3.2.
# Report any issues to https://github.com/b-rodrigues/rix
let
 pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/219f896bdf192721446db4fedc338eebf732057d.tar.gz") {};
     
  system_packages = builtins.attrValues {
    inherit (pkgs) 
      R
      glibcLocales
      nix;
  };
  
in

pkgs.mkShell {
  LOCALE_ARCHIVE = if pkgs.system == "x86_64-linux" then  "${pkgs.glibcLocales}/lib/locale/locale-archive" else "";
  LANG = "en_US.UTF-8";
   LC_ALL = "en_US.UTF-8";
   LC_TIME = "en_US.UTF-8";
   LC_MONETARY = "en_US.UTF-8";
   LC_PAPER = "en_US.UTF-8";
   LC_MEASUREMENT = "en_US.UTF-8";

  buildInputs = [    system_packages   ];
  
}

```

**Comment by philipp-baumann** [comment]
It works like this:

```sh
# This file was generated by the {rix} R package v0.7.1 on 2024-06-24
# with following call:
# >rix(r_ver = "219f896bdf192721446db4fedc338eebf732057d",
#  > r_pkgs = NULL,
#  > system_pkgs = NULL,
#  > git_pkgs = NULL,
#  > ide = "rstudio",
#  > project_path = "_rstudio",
#  > overwrite = TRUE,
#  > print = TRUE,
#  > shell_hook = "export QT_XCB_GL_INTEGRATION=none")
# It uses nixpkgs' revision 219f896bdf192721446db4fedc338eebf732057d for reproducibility purposes
# which will install R version 4.3.2.
# Report any issues to https://github.com/b-rodrigues/rix
let
 pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/219f896bdf192721446db4fedc338eebf732057d.tar.gz") {};
     
  system_packages = builtins.attrValues {
    inherit (pkgs) 
      R
      glibcLocales
      nix
      rstudio;
  };
  
in

pkgs.mkShell {
  LOCALE_ARCHIVE = if pkgs.system == "x86_64-linux" then  "${pkgs.glibcLocales}/lib/locale/locale-archive" else "";
  LANG = "en_US.UTF-8";
   LC_ALL = "en_US.UTF-8";
   LC_TIME = "en_US.UTF-8";
   LC_MONETARY = "en_US.UTF-8";
   LC_PAPER = "en_US.UTF-8";
   LC_MEASUREMENT = "en_US.UTF-8";

  buildInputs = [    system_packages   ];
  shellHook = "export QT_XCB_GL_INTEGRATION=none";
}
```

it is quite a special case when `r_pkgs = NULL`, but if there is IDE rstudio, we need to add rstudio to `system_packages` instead of going via the rstudioWrapper approach.

**Comment by philipp-baumann** [comment]
@b-rodrigues i wonder whether we should generally add `shell_hook =  "export QT_XCB_GL_INTEGRATION=none"` when selecting rstudio (seems in Ubuntu and OpenSuse, do we have systematic tests for other linux flavors? it seems also like something that needs to be addressed in nix rstudio packaging in nixpkgs.) What do you think?

**Comment by philipp-baumann** [comment]
![libpaths-bug-rstudio](https://github.com/b-rodrigues/rix/assets/21625034/928ff288-6ce5-4495-b8b4-e2fd4cdc78a4)

we have another bug, but this is quite for sure on the rstudio side. .libPaths() should not show `R_LIBS_USER` from the system anymore. It seems RStudio is still modifying .libPaths() after sourcing local .Rprofile, which is not great.

**Comment by b-rodrigues** [comment]
> @b-rodrigues i wonder whether we should generally add `shell_hook = "export QT_XCB_GL_INTEGRATION=none"` when selecting rstudio (seems in Ubuntu and OpenSuse, do we have systematic tests for other linux flavors? it seems also like something that needs to be addressed in nix rstudio packaging in nixpkgs.) What do you think?

Just tried, we don't even need to put that as a shellHook, we could detect any linux (perhaps only "real" linuxes" not WSL) and then add it like this:

```
pkgs.mkShell {
  LOCALE_ARCHIVE = if pkgs.system == "x86_64-linux" then  "${pkgs.glibcLocales}/lib/locale/locale-archive" else "";
  LANG = "en_US.UTF-8";
   LC_ALL = "en_US.UTF-8";
   LC_TIME = "en_US.UTF-8";
   LC_MONETARY = "en_US.UTF-8";
   LC_PAPER = "en_US.UTF-8";
   LC_MEASUREMENT = "en_US.UTF-8";

   QT_XCB_GL_INTEGRATION="none";

  buildInputs = [  rpkgs  system_packages  wrapped_pkgs ];
  
}

```

This should probably work even if not required? How does it work on your rocky linux?

**Comment by b-rodrigues** [comment]
I think we shouldn't allow ide="rstudio" and no packages to be added at all, see this PR: https://github.com/b-rodrigues/rix/pull/228

**Comment by philipp-baumann** [comment]
Closing, resolved with #228
